core/Http/RequestSanitizer.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Http;
+
+/**
+ * Service that sanitizes query parameters for requests.
+ *
+ * Eventually once consistent out-only sanitization is implemented, this class should be
+ * removed. In the mean time, since this class is accessed through DI, it can be used
+ * to implement consistent sanitization, piece by piece.
+ */
+class RequestSanitizer
+{
+    // Flag used with htmlspecialchar. See php.net/htmlspecialchars.
+    const HTML_ENCODING_QUOTE_STYLE = ENT_QUOTES;
+
+    /**
+     * @var array[]
+     */
+    private $requestParameterSanitizeBlacklist = array();
+
+    public function __construct(array $requestParameterSanitizeBlacklist)
+    {
+        foreach ($requestParameterSanitizeBlacklist as $entry) {
+            list($apiMethod, $params) = $entry;
+
+            if (isset($this->requestParameterSanitizeBlacklist[$apiMethod])) {
+                $this->requestParameterSanitizeBlacklist[$apiMethod] =
+                    array_merge($this->requestParameterSanitizeBlacklist[$apiMethod], $params);
+            } else {
+                $this->requestParameterSanitizeBlacklist[$apiMethod] = $params;
+            }
+        }
+    }
+
+    /**
+     * Sanitizes a single value.
+     *
+     * @param mixed $value If numeric, the value is not changed. If a string, line breaks are removed and HTML entities
+     *                     are escaped.
+     * @param bool $alreadyStripslashed Implementation detail, ignore.
+     * @return string
+     * @throws \Exception If the value is null or a boolean.
+     */
+    public function sanitizeValue($value, $alreadyStripslashed = false)
```

Comment:
I can make this change. Note: this class should be removed completely for 3.0, so the names shouldn't matter too much.